library/std/src/sys/windows/fs.rs

```diff
@@ -1062,7 +1065,17 @@ fn remove_dir_all_iterative(f: &File, delete: fn(&File) -> io::Result<()>) -> io
         // If there were no more files then delete the directory.
         if !more_data {
             if let Some(dir) = dirlist.pop() {
-                delete(&dir)?;
+                // Retry deleting a few times in case we need to wait for a file to be deleted.
+                for i in 1..=MAX_RETRIES {
+                    let result = delete(&dir);
+                    if let Err(e) = result {
+                        if i == MAX_RETRIES || e.kind() != io::ErrorKind::DirectoryNotEmpty {
+                            return Err(e);
+                        }
+                    } else {
+                        break;
+                    }
```

Comment:
This usually is due to AV, right? Does a `thread::yield_now` do anything on windows that would make it more likely that the other program would get to finish its work?

Reply:
Usually due to AV but IIRC cargo have also had problems with trying to delete a directory containing an exe too quickly after the process returns.

This does indeed seem to be a case where `thread::yield_now` may be helpful. Or at the very least not harmful.